src/common/Cryptography/BigNumber.cpp

```diff
@@ -185,15 +179,14 @@ bool BigNumber::IsNegative() const
     return BN_is_negative(_bn);
 }
 
-bool BigNumber::AsByteArray(uint8* buf, std::size_t bufsize, bool littleEndian) const
+void BigNumber::GetBytes(uint8* buf, std::size_t bufsize, bool littleEndian) const
 {
     int nBytes = GetNumBytes();
-    ASSERT(!(nBytes < 0));
+    ASSERT(!(nBytes < 0), "Bignum has negative number of bytes (%d).", nBytes);
```

Comment:
```suggestion
    ASSERT(nBytes >= 0, "Bignum has negative number of bytes (%d).", nBytes);
```
if you like. I find it more readable.